### PR TITLE
Implement gRPC GetArtifactType

### LINKED ISSUE #25: Implement gRPC `GetArtifactType`
per title.

### PR DESCRIPTION
**Pull request by tarilabs**
resolves #25 

## Description
Implement gRPC `GetArtifactType`

## How Has This Been Tested?
in one terminal:

```
make build
make run/server
```

in another terminal:

```
python test/python/test_mlmetadata.py 
```

then running the following python script:

```python
from grpc import insecure_channel
from ml_metadata.proto import metadata_store_service_pb2
from ml_metadata.proto import metadata_store_service_pb2_grpc

def main():
    channel = insecure_channel('localhost:8080')
    store = metadata_store_service_pb2_grpc.MetadataStoreServiceStub(channel)

    request = metadata_store_service_pb2.GetArtifactTypeRequest()
    request.type_name = "SavedModel"

    response = store.GetArtifactType(request)
    print(response)

if __name__ == '__main__':
    main()
```

Outputs as expected:

```
$ python test/python/test_getartifacttype.py 
artifact_type {
  id: 2
  name: "SavedModel"
}
```

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: internal/server/grpc_server.go [hunk 1]
```diff
@@ -268,9 +269,38 @@ func (g grpcServer) PutParentContexts(ctx context.Context, request *proto.PutPar
 	panic("implement me")
 }
 
-func (g grpcServer) GetArtifactType(ctx context.Context, request *proto.GetArtifactTypeRequest) (*proto.GetArtifactTypeResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (g grpcServer) GetArtifactType(ctx context.Context, request *proto.GetArtifactTypeRequest) (resp *proto.GetArtifactTypeResponse, err error) {
+	ctx, dbConn := Begin(ctx, g.dbConnection)
+	defer handleTransaction(ctx, &err)
```
Comment (tarilabs) [inline internal/server/grpc_server.go:271]: I preferred to take "same approach" and place this on a tx context too, even if this is only a read, for consistency.

**Review by dhirajsb — CHANGES_REQUESTED** [review]
Please add the test client code to the current prototype integration test `test/python/test_mlmetadata.py` script. We can re-use it in a later testframework.

**Review by lampajr — DISMISSED** [review]

**Review by lampajr — APPROVED** [review]

**Review by dhirajsb — APPROVED** [review]
:+1:

**Review by antisaling — APPROVED** [review]
lgtm!

**Comment by tarilabs** [comment]
no vetos, no missing, all approval; proceeding to merge.

**Pull request merged**